### PR TITLE
Add optional type

### PR DESCRIPTION
**Pull request by ogaca-dd**
### What does this PR do?

Ths PR introduces an generic  `Optional` type.
 
### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by pgimalac — APPROVED** [review]
LGTM, maybe a `NewNone[T]()` constructor could be clearer than `Optional[T]{}` but I don't have strong feelings about that.

File: pkg/util/optional.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package util
+
+// Optional reprensents an optional type.
```
Comment (pgimalac) [inline pkg/util/optional.go:8]:
```suggestion
// Optional represents an optional type.
```

**Review by pgimalac — APPROVED** [review]
Just remembered that I saw a typo :)

File: pkg/util/optional.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+// By default, no value is set and a call to GetValue() returns (T{}, false)
+type Optional[T any] struct {
+	value T
+	set   bool
+}
+
+// NewOptional creates a new Optional[T] type with a value. A call to GetValue() returns (value, true)
+func NewOptional[T any](value T) Optional[T] {
```
Comment (gh123man) [inline pkg/util/optional.go:16]: Should we also provide a `Set` and possibly `UnSet` methods if we want to change the optional value after initialization?

File: pkg/util/optional.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+	o.value = value
+	o.set = true
+}
+
+// Reset removes the value set.
+func (o *Optional[T]) Reset() {
+	o.set = false
+}
+
+// MapOptional returns fct(value) is a value is set, otherwise it returns NewNoneOptional[T2]().
```
Comment (pgimalac) [inline pkg/util/optional.go:44]:
```suggestion
// MapOptional returns fct(value) if a value is set, otherwise it returns NewNoneOptional[T2]().
```

**Review by pgimalac — APPROVED** [review]

**Review by gh123man — APPROVED** [review]

**Pull request merged**